File: tests/fixtures/sample.go
// Sample Go file for testing tree-sitter parsing.
package sample

import (
	"fmt"
	"os"
)

// Greeter interface for greeting behavior.
type Greeter interface {
	Greet(name string) string
}

// HelloGreeter implements Greeter.
type HelloGreeter struct {
	Prefix string
}

// Greet returns a greeting message.
func (h *HelloGreeter) Greet(name string) string {
	return fmt.Sprintf("%s, %s!", h.Prefix, name)
}

// standalone function
func ProcessData(data []int) int {
	sum := 0
	for _, v := range data {
		if v > 0 {
			for i := 0; i < v; i++ {
				sum += i
			}
		}
	}
	return sum
}

func main() {
	g := &HelloGreeter{Prefix: "Hello"}
	fmt.Println(g.Greet("World"))
	os.Exit(0)
}

File: tests/fixtures/sample.rs
// Sample Rust file for testing tree-sitter parsing.

use std::fmt;
use std::io::{self, Write};

pub trait Greeter {
    fn greet(&self, name: &str) -> String;
}

pub struct HelloGreeter {
    prefix: String,
}

impl HelloGreeter {
    pub fn new(prefix: &str) -> Self {
        HelloGreeter {
            prefix: prefix.to_string(),
        }
    }
}

impl Greeter for HelloGreeter {
    fn greet(&self, name: &str) -> String {
        format!("{}, {}!", self.prefix, name)
    }
}

pub enum Status {
    Active,
    Inactive,
    Pending(u32),
}

fn process_data(data: &[i32]) -> i32 {
    let mut sum = 0;
    for &v in data {
        if v > 0 {
            for i in 0..v {
                sum += i;
            }
        }
    }
    sum
}

fn main() {
    let g = HelloGreeter::new("Hello");
    println!("{}", g.greet("World"));
}
